db/version_set.cc

```diff
@@ -1863,6 +1864,118 @@ Status Version::GetBlob(const ReadOptions& read_options, const Slice& user_key,
   return s;
 }
 
+void Version::MultiGetBlob(const ReadOptions& read_options,
+                           MultiGetRange& range) {
+  // Guaranteed by caller.
+  assert(range.NeedsAnyBlobRead());
+
+  if (read_options.read_tier == kBlockCacheTier) {
+    Status s = Status::Incomplete("Cannot read blob(s): no disk I/O allowed");
+    for (auto it = range.blob_begin(); it != range.blob_end(); ++it) {
+      assert(it->s->ok());
+      *(it->s) = s;
+    }
+    return;
+  }
+
+  using BlobIterator = MultiGetRange::BlobIterator;
+  std::unordered_map<uint64_t, std::vector<std::pair<BlobIndex, BlobIterator>>>
+      blob_rqs;
```

Comment:
I'm wondering if we built this map in `Version::MultiGet` as we're encountering the blob references, would that eliminate the need for the new mask and related logic in `MultiGetContext`?

Reply:
Done.
The benefit of an extra mask `blob_mask_` is that we can conveniently iterate over the multiget requests and locate individual requests that needs blob reads. It can be removed, though.